### ISSUE TITLE
Recent change broke Offset Protocol

### ISSUE THREAD

**Issue by Crim** [nmred/kafka-php#35]
So the following line change: https://github.com/nmred/kafka-php/commit/12d4f2f4fc487e5629d5fd8dcdc5ff4329123427#diff-6a05497ce6c49eaedc44b9b2b7075c8eL69

Broke retrieving offsets from Kafka.  The response I get is:
array(1) { ["MyTopic"]=> array(1) { [0]=> array(3) { ["offset"]=> int(-1) ["metadata"]=> string(0) "" ["errCode"]=> int(768) } } } 

Reverting that line and I can correctly retrieve offsets for various topics/partitions via the Offset api, which looks like
array(1) { ["MyTopic"]=> array(1) { [0]=> array(3) { ["offset"]=> int(-1) ["metadata"]=> string(0) "" ["errCode"]=> int(3) } } }

**Comment by Crim** [comment]
@AlexKR --^

**Comment by AlexKR** [comment]
@Crim thanks for issue. looking very strangely, i will try investigate more about it and will commit correct way for error detecting

**Comment by Crim** [comment]
For reference this is running on a x64 machine against kafka 0.8.2.1.  Also I didn't test the other code path that has the same change, so not sure if that has issues or not.

**Comment by Crim** [comment]
So I took a stab at this.  Looking at PHP Docs for pack here: http://php.net/manual/en/function.pack.php

We were using 'n' to unpack this which says: unsigned short (always 16 bit, big endian byte order)
It got swapped to 's' which says: signed short (always 16 bit, machine byte order)

I'm guessing that your system @AlexKR is big endian, where as mine is little endian, which is why I assume this works for you, but not me.

I opened this PR which attempts to detect the endianess of the system the code is running on, and flips the endianess after unpacking to match and be consistent on either platform.

We/I should probably write a test or two to cover this before it gets merged in tho.

https://github.com/nmred/kafka-php/pull/36
